pinery-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java

```diff
@@ -0,0 +1,1031 @@
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.sql.DataSource;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.text.WordUtils;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+
+import com.google.common.collect.Lists;
+
+import ca.on.oicr.pinery.api.Attribute;
+import ca.on.oicr.pinery.api.AttributeName;
+import ca.on.oicr.pinery.api.Box;
+import ca.on.oicr.pinery.api.Change;
+import ca.on.oicr.pinery.api.ChangeLog;
+import ca.on.oicr.pinery.api.Instrument;
+import ca.on.oicr.pinery.api.InstrumentModel;
+import ca.on.oicr.pinery.api.Lims;
+import ca.on.oicr.pinery.api.Order;
+import ca.on.oicr.pinery.api.OrderSample;
+import ca.on.oicr.pinery.api.PreparationKit;
+import ca.on.oicr.pinery.api.Run;
+import ca.on.oicr.pinery.api.RunPosition;
+import ca.on.oicr.pinery.api.RunSample;
+import ca.on.oicr.pinery.api.Sample;
+import ca.on.oicr.pinery.api.SampleProject;
+import ca.on.oicr.pinery.api.Status;
+import ca.on.oicr.pinery.api.Type;
+import ca.on.oicr.pinery.api.User;
+import ca.on.oicr.pinery.lims.DefaultAttribute;
+import ca.on.oicr.pinery.lims.DefaultAttributeName;
+import ca.on.oicr.pinery.lims.DefaultChangeLog;
+import ca.on.oicr.pinery.lims.DefaultInstrument;
+import ca.on.oicr.pinery.lims.DefaultInstrumentModel;
+import ca.on.oicr.pinery.lims.DefaultOrder;
+import ca.on.oicr.pinery.lims.DefaultPreparationKit;
+import ca.on.oicr.pinery.lims.DefaultRun;
+import ca.on.oicr.pinery.lims.DefaultSample;
+import ca.on.oicr.pinery.lims.DefaultSampleProject;
+import ca.on.oicr.pinery.lims.DefaultStatus;
+import ca.on.oicr.pinery.lims.DefaultType;
+import ca.on.oicr.pinery.lims.DefaultUser;
+import ca.on.oicr.pinery.lims.miso.MisoClient.SampleRowMapper.AttributeKey;
+import ca.on.oicr.pinery.lims.miso.converters.SampleTypeConverter;
+
+public class MisoClient implements Lims {
+
+  private static final String MISO_SAMPLE_ID_PREFIX = "SAM";
+  private static final String MISO_LIBRARY_ID_PREFIX = "LIB";
+  private static final String MISO_DILUTION_ID_PREFIX = "LDI";
+
+  // @formatter:off
+  // InstrumentModel queries
+  private static final String queryAllModels = "SELECT p.platformId, p.instrumentModel " + "FROM Platform as p";
+  private static final String queryModelById = queryAllModels + " WHERE p.platformId = ?";
+
+  // Instrument queries
+  private static final String queryAllInstruments = "SELECT i.instrumentId, i.name, i.platformId " + "FROM Instrument AS i";
+  private static final String queryInstrumentById = queryAllInstruments + " WHERE i.instrumentId = ?";
+  private static final String queryInstrumentsByModelId = queryAllInstruments + " WHERE i.platformId = ?";
+
+  // Order queries
+  private static final String queryAllOrders = getResourceAsString("queryAllOrders.sql");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)